workshops/node_intro/README.md

```diff
@@ -0,0 +1,288 @@
+It makes any files in `public` available to the frontend when it searches for them.
+```js
+// give them access to everything in /public
+app.use(express.static(__dirname + '/public')); 
+```
+
+After that snippet of code, we send anyone who visits our site `index.html`, as I've already explained.  
+`/` is the root of your application.  
+You could use fancier Express code than this to send a special page
+to anyone who goes to or is redirected to the `workshop/` directory of `localhost`, for example.
+Naturally, there's a lot more to it than that, Express is pretty powerful.
+```js
+app.get('/', function(req, res, next) {
+    res.sendFile(__dirname + '/public/index.html');
+});
+```
+
+Second to last in our tiny Express example is this block of code.
+```js
+// express error handling
+app.use(function logErrors (err, req, res, next) {
+    console.error(err.stack);
+    next(err);
+});
+```
+Sometimes in Express, things go wrong. This just funnels those errors right into your console.
+Perhaps if you were running your server in production and wanted a better idea of what was going on even if your console wasn't nearby,
+you could have the errors logged into a file you could read at your leisure.
+
+
+And finally, there's the listen statement, where you can provide your port.
+```js
+app.listen(8000);
+```
+Note that no server is actually hosted until this line is called.
+Right now, it's run at the bottom of the file.
+It doesn't have to be.
+For example you may want to load a bunch of assets into the server's memory first
+before you start to allow connections.
+
+### Shoutout to Nodemon
+If you're familiar with Repl.it or Glitch as many beginners are, you're probably spoiled on their live reloading functionality.
+There's no shame in that, I'm addicted to live reloading as well.  
+https://nodemon.io/  
+You can easily install this command line utility using a single command, and then have your server automatically reload as soon as any changes are detected.
+The key difference is that you start your app with just `nodemon` instead of `node index`.
+
+# Wrapup.
+This is the end of my tiny crash course example on Node and Express. There's a lot more to learn about either in their respective documentation.
+Express has a ton of awesome content for writing more complex web apps, and I would encourage you to dive into their documentation to learn more.  
+https://expressjs.com/  
+https://nodejs.org/en/  
+
+Now you have a solid webserver you can connect to to get a webpage from.
+You could use this to run a game or a simple website.
+You now know the core concepts and terminology of the server/client architecture that the web is built on.
+If you hear someone refer to themselves as a "Frontend" or "Backend" dev, you'll know what they mean.
+(If they say "Full Stack", they mean they do both.)
+Along the way, you also learned some terminal commands and how to start a new Node.js project.
```

Comment:
Might be worth noting that you can use Express in conjunction with other frameworks? (e.g. React, Django)